### PR TITLE
Recipe collection

### PR DESCRIPTION
**Pull request by kawrou**
Recipe collection should be ready. But will need changing because of update to state sharing in app. 
Creating a draft pull request for now.

### CODE REVIEW

**Comment by kawrou** [comment]
A few of the tests are broken. I've fixed:
RecipeCollection Page
useFetchRecipe
RecipeScraper

Need to fix other tests on different branch to not hold up this PR

I've tried to improve on same Expect / Matchers so hopefully there would be less breaking.

**Comment by kawrou** [comment]
Fixed recipes.test for services/recipes

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -94,7 +94,18 @@ const App = () => {
           path="/recipes/:recipe_id"
           element={<SingleRecipePage token={token} setToken={setToken} />}
         />
-        <Route path="/recipecollection" element={<RecipeCollection />} />
+        <Route
+          path="/recipecollection"
```
Comment (mattwsheps) [inline frontend/src/App.jsx]: Change path to '/myrecipes'

**Review by mattwsheps — COMMENTED** [review]
Looks good other than potential path name change and maybe a name change for RecipeCollection.jsx to MyRecipesPage.jsx?

File: frontend/src/App.jsx (hunk 1 above)
Comment (kawrou) [reply frontend/src/App.jsx]: > Change path to '/myrecipes'

Done!

**Comment by kawrou** [comment]
Ok, I think I'm really finished with it now. 
I've renamed RecipeCollection
I've deleted GetRecipes service as deprecated.
The useFetchRecipes custom hook will now use the proper function inside of services/recipes

I've added TODO: in the test files for where I think it will need more test coverage later or where I think the mocking isn't quite right. 

I've fixed the test in HomePage but HomePage needs some work! 
It's just to verify that a user would be navigated to a login or createRecipe page.

If this is merged first thing, then can merge the backend which hopefully will be quick ...

**Review by gabrielaehrenbrink — APPROVED** [review]
Looks good!

**Review by ben-dh3 — APPROVED** [review]

**Pull request merged**